pandas/tests/indexes/base_class/test_indexing.py

```diff
@@ -76,3 +76,11 @@ def test_get_loc_nan_object_dtype_nonmonotonic_nonunique(self):
         # we don't match at all on mismatched NA
         with pytest.raises(KeyError, match="NaT"):
             idx.get_loc(NaT)
+
+
+def test_getitem_boolean_ea_indexer():
+    # GH#45806
+    ser = pd.Series([True, False, pd.NA], dtype="boolean")
+    result = ser.index[ser]
+    expected = Index([0])
```

Comment:
is there a discussion where it was decided that this is the correct behavior?  could plausibly raise.

could also use ser._values or Index(ser) for the key/

Reply:
This works for a regular bool series, so thought that this should work too. We can add these test cases too, but if Series should work we have to keep that one

Reply:
@phofl just so im clear, your comment is responding to the "could also use..." part of mine?  If so that's fine.  I'm unclear on the other part of why we're not raising on pd.NA

Reply:
> Why should we raise on pd.NA?

I think of `idx[mask]` as akin to `[idx[n] for n in range(len(mask)) if mask[n]]` which would raise.